### DISCUSSION TITLE
MSP creation using Pure Wix method - torch.exe

### DISCUSSION THREAD

**Discussion by yuvnith** [wixtoolset/issues#7651] [Questions]
Hi, I am trying to create an MSP using pure Wix method. I am facing an issue because of renaming functionality in my MSI. Below is my MSI code.

<Component Id ="EnglishComponent" Guid ="Guid" Directory ="Directory">
        <File Id ="DLLEnglish" Source ="LanguageFile.dll" KeyPath ="yes"/>
</Component>

 <Component Id ="FrenchComponent" Guid ="Guid2" Directory ="Directory">
        <File Id ="DLLFrench" Source ="FrenchLanguageFile.dll" KeyPath ="yes" Name="LanguageFile.dll"/>
        <Condition><![CDATA[CULTUREINFO << "fr-"]]></Condition>
</Component>

We extract the OS culture and use it within the condition, and copy the appropriate localization dlls. Renaming is also done after copying  (to LanguageFile.dll)
I have similar MSI's for both RTM and Hot Fix, the only difference is that, HF MSI has new dll versions. 

Steps executed to create MSP:
---------------------------------------
Ran the msiexec for both RTM and HF msi, and fed the resultant files to Torch.exe command (MSI and extracted files)  
Then executed Candle.exe, Light.exe and Pyro.exe commands to create an MSP 

When we run the HF5 on an English OS, observed that EnglishComponent was picked up but wrong file was copied i.e. French localization file with name LanguageFile.dll

Assuming this happened because, while running exec step, it first copied English dll with name LanguageFile.dll and then French dll (renamed to LanguageFile.dll after copy - overwriting existing English dll) Then, the French language dll (with name is LanguageFile.dll) is mapped to all the components (EnglishComponent and FrenchComponent)

**Comment by robmen** [comment]
There is not enough detail here to really be useful. You might start by providing much more detail about each step you've done and what you see with the patch applied and not applied in Orca.
